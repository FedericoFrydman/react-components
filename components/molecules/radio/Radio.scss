@import '../../assets/styles/colors.scss';

.lsux-rb-container {
  align-items: center;
  display: flex;

  input {
    height: 0;
    margin: 0;
    opacity: 0;
    width: 0;
  }

  &__rb {
    border-radius: 50%;
    cursor: pointer;
    height: 20px;
    width: 20px;

    &.margin-left {
      margin-left: 8px;
    }

    &.margin-right {
      margin-right: 8px;
    }
  }

  input ~ &__rb {
    border: 2px solid $N600;
  }

  body.using-key input:focus ~ &__rb {
    outline: 2px solid $N100;
  }

  input:checked ~ &__rb {
    background: radial-gradient(var(--radio-cb-enabled-checked-background) 45%, #fff 46.2%, transparent 0%);
    border: 2px solid var(--radio-cb-enabled-checked-border);
  }

  input:disabled ~ span {
    color: var(--radio-cb-disabled-checked-border);
  }

  input:disabled ~ &__rb {
    background-color: var(--radio-cb-disabled-unchecked-background);
    border: 1px solid var(--radio-cb-disabled-unchecked-border);
    cursor: not-allowed;
  }

  input:disabled:checked ~ &__rb {
    background-color: var(--radio-cb-disabled-checked-background);
    border: 5px solid var(--radio-cb-disabled-checked-background);
  }
}
