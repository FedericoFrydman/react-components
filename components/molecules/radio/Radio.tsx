import { Ref, forwardRef } from 'react';
import cn from 'classnames';

import './Radio.scss';

export interface RadioProps extends Omit<React.HTMLProps<HTMLInputElement>, 'css'> {
  /**
   * Label to the right of the button
   */
  rightLabel?: string;
  /**
   * Label to the left of the button
   */
  leftLabel?: string;
  /**
   * Custom classes for the component.
   */
  classNames?: cn.Argument;
  /**
   * The ref to the HTML DOM element.
   */
  ref?: Ref<HTMLInputElement>;
}

const RadioComponent = ({ rightLabel, leftLabel, classNames = [], ...props }: RadioProps, ref: RadioProps['ref']) => {
  const classContainer = 'lsux-rb-container';
  const cnContainer = cn(
    {
      [`${classContainer}`]: true,
    },
    classNames,
  );

  const classContainerRb = `${classContainer}__rb`;
  const cnContainerRb = cn({
    [`${classContainerRb}`]: true,
    ['margin-left']: leftLabel,
    ['margin-right']: rightLabel,
  });

  return (
    <label className={cnContainer}>
      <input {...props} ref={ref} type="radio" />
      {leftLabel && <span>{leftLabel}</span>}
      <div className={cnContainerRb}></div>
      {rightLabel && <span>{rightLabel}</span>}
    </label>
  );
};

/**
 * A radio button is a quick way to extract a single coded answer from our users by presenting multiple possibilities and allowing only one option to be chosen.
 */

export const Radio = forwardRef<HTMLInputElement, RadioProps>(RadioComponent);
