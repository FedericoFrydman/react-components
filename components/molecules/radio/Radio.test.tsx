import { shallow, ShallowWrapper } from 'enzyme';
import { Radio } from './Radio';

describe('Radio Component', () => {
  let testId: string;
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    testId = 'test-radio';
    wrapper = shallow(<Radio name={testId} />);
  });

  test('should render the component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
