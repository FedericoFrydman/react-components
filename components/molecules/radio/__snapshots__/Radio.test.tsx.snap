// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Component should render the component 1`] = `
<label
  className="lsux-rb-container"
>
  <input
    name="test-radio"
    type="radio"
  />
  <div
    className="lsux-rb-container__rb"
  />
</label>
`;
